device/Microsoft.Azure.Devices.Client/TransientFaultHandling/AsyncExecution.cs

```diff
@@ -65,7 +73,7 @@ private static Task<bool> StartAsGenericTask(Func<Task> taskAction)
                 }), "taskAction");
             }
             TaskCompletionSource<bool> tcs = new TaskCompletionSource<bool>();
-            task.ContinueWith(delegate (Task t)
+            await task.ContinueWith(delegate (Task t)
```

Comment:
This is very odd - the code is configuring a task continuation then you are awaiting it?
It's basically the same thing as writing 
```C# 
await task; 
await continuation;
```

I don't think this was the intent here. Continuations are the right thing to use: the problem is when we retry, we should _not_ append a continuation, rather we should complete the original task and start another one.